docs/configuration.md

```diff
@@ -0,0 +1,162 @@
+# Configuration
+
+Operator configuration can be modified by editing the SSP Custom Resource (CR).
+
+## Table Of Contents
+
+- [SSP Custom Resource (CR)](#ssp-custom-resource-cr)
+- [Annotations](#annotations)
+- [Common Templates](#common-templates)
+- [Template Validator](#template-validator)
+- [Tekton Pipelines](#tekton-pipelines)
+- [Tekton Tasks](#tekton-tasks)
+- [Feature Gates](#feature-gates)
+
+## SSP Custom Resource (CR)
+
+To activate the operator, create the SSP Custom Resource (CR):
+```
+apiVersion: ssp.kubevirt.io/v1beta2
+kind: SSP
+metadata:
+  annotations:
+    ssp.kubevirt.io/vm-console-proxy-enabled: "true"
+    ssp.kubevirt.io/vm-console-proxy-namespace: "kubevirt"
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  commonTemplates:
+    namespace: kubevirt
+  templateValidator:
+    replicas: 2
+  featureGates:
+    deployTektonTaskResources: true
+  tektonPipelines:
+    namespace: kubevirt
+  tektonTasks:
+    namespace: kubevirt
+```
+
+## Annotations
+
+### VM Console Proxy
+
+These annotations are used by VM console proxy operand.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  annotations:
+    ssp.kubevirt.io/vm-console-proxy-enabled: "true" # If not set, then by default is false
+    ssp.kubevirt.io/vm-console-proxy-namespace: "kubevirt" # If not set, then default namespace is "kubevirt"
+  name: ssp-sample
+  namespace: kubevirt
+spec: {}
+```
+
+### Pause Operator
+
+This annotation will pause operator reconciliation.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  annotations:
+    kubevirt.io/operator.paused: "false" # If not set, then by default is false
+  name: ssp-sample
+  namespace: kubevirt
+spec: {}
+```
+
+The operator will not react to any changes to the `SSP` resource
+or any of the watched resources. If a paused `SSP` resource is deleted,
+the operator will still cleanup all the dependent resources.
+
+## Common Templates
+
+A set of common templates to create KubeVirt Virtual Machines (VMs).
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  commonTemplates:
+    namespace: kubevirt
+```
+
+## Template Validator
+
+Validating webhook to check templates sent to the cluster.
+
+```
+apiVersion: ssp.kubevirt.io/v1beta1
+kind: SSP
+metadata:
+  name: ssp-sample
+  namespace: kubevirt
+spec:
+  templateValidator:
+    replicas: 2
```

Comment:
Probably worth to mention, that user can configure number of replicas for validator deployment (in other paragraphs, you are mentioning what user can set)